scijava-ops-image/src/main/java/org/scijava/ops/image/coloc/saca/WtKendallTau.java

```diff
@@ -0,0 +1,236 @@
+/*-
+ * #%L
+ * ImageJ2 software for multidimensional image processing and analysis.
+ * %%
+ * Copyright (C) 2014 - 2024 ImageJ2 developers.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+
+package org.scijava.ops.image.coloc.saca;
+
+import java.lang.Double;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.Random;
+
+import org.scijava.ops.image.coloc.IntComparator;
+
+/**
+ * Helper class for Spatially Adaptive Colocalization Analysis (SACA) framework.
+ * This class is used by the AdaptiveSmoothedKendallTau class.
+ *
+ * @author Shulei Wang
+ * @author Ellen TA Dobson
+ * @author Curtis Rueden
+ */
+
+final class WtKendallTau {
```

Comment:
Could also consider making these their own ops

Reply:
This one I feel less strongly about, because I do not know whether these would be used elsewhere

Reply:
I'm against making the `WtKendallTau.java` helper class an Op. It calculates the weighted [`kendall tau rank correlation`](https://en.wikipedia.org/wiki/Kendall_rank_correlation_coefficient) that determines the level/strength of colocalization within a set of data. My understanding here is that this kendall tau class is a specific implementation for SACA and would require a little bit of work to generalize. For now I think this is okay as is.